src/tap/cmd.rs

```diff
@@ -70,20 +80,30 @@ pub async fn cmd(opts: &super::Opts, mut signal_rx: SignalRx) -> exitcode::ExitC
         let stream = res.stream();
     };
 
-    // Loop over the returned results, printing out log events.
-    // NOTE: This will currently ignore notifications. A later `--verbose` option is planned
-    // to include these.
-    // TODO: https://github.com/vectordotdev/vector/issues/6870
+    // Loop over the returned results, printing out tap events.
     loop {
         tokio::select! {
             biased;
             Some(SignalTo::Shutdown | SignalTo::Quit) = signal_rx.recv() => break,
             Some(Some(res)) = stream.next() => {
                 if let Some(d) = res.data {
-                    for event_string in d.output_events_by_component_id_patterns.iter().filter_map(|ev| ev.as_string()) {
-                        #[allow(clippy::print_stdout)]
-                        {
-                            println!("{}", event_string);
+                    for tap_event in d.output_events_by_component_id_patterns.iter() {
+                        match tap_event {
+                            OutputEventsByComponentIdPatternsSubscriptionOutputEventsByComponentIdPatterns::Log(ev) => {
+                                println!("{}", ev.string);
+                            },
+                            OutputEventsByComponentIdPatternsSubscriptionOutputEventsByComponentIdPatterns::Metric(ev) => {
+                                println!("{}", ev.string);
+                            },
+                            OutputEventsByComponentIdPatternsSubscriptionOutputEventsByComponentIdPatterns::EventNotification(ev) => {
+                                if !opts.quiet {
+                                    match ev.notification {
+                                        EventNotificationType::MATCHED => eprintln!(r#"[tap] Pattern "{}" successfully matched."#, ev.pattern),
```

Comment:
(Not a blocker) - One idea I had for #6898 was to call out meta with some kind of terminal styling -- making the text bold, or modifying the colour.

That might be useful as a follow-up to give users some at-a-glance differentiation between control messages and data.

Reply:
Ah, that makes sense, noted. Will look into this.